### PR TITLE
Fix the location of tmpfs pwrite write and Fixed some code check warnings

### PR DESCRIPTION
**Pull request by Zhangshoukui**
## Summary
1. Fix the location of tmpfs pwrite write
2. Fixed some code check warnings
## Impact
tmpfs
## Testing
yes

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@Zhangshoukui please remove Change-Id from commit message

**Review by xiaoxiang781216 — APPROVED** [review]

File: fs/tmpfs/fs_tmpfs.c [hunk 1]
```diff
@@ -1605,7 +1605,15 @@ static ssize_t tmpfs_write(FAR struct file *filep, FAR const char *buffer,
 
   /* Handle attempts to write beyond the end of the file */
 
-  startpos = filep->f_pos;
+  if ((filep->f_oflags & O_APPEND) == O_APPEND)
```
Comment (anchao) [inline fs/tmpfs/fs_tmpfs.c:1608]:
```suggestion
  if ((filep->f_oflags & O_APPEND) != 0)
```

**Review by anchao — CHANGES_REQUESTED** [review]

File: fs/tmpfs/fs_tmpfs.c (hunk 1 above)
Comment (Zhangshoukui) [reply fs/tmpfs/fs_tmpfs.c]: done

Comment (anchao) [reply fs/tmpfs/fs_tmpfs.c]: Thanks, check 0 is faster than check constant

**Review by anchao — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]
please add a commit message describing the change

**Pull request merged**